api/src/transactions.rs

```diff
@@ -1365,12 +1366,24 @@ impl TransactionsApi {
 
         // Simulate transaction
         let state_view = self.context.latest_state_view_poem(&ledger_info)?;
-        let (vm_status, output) =
+        let (vm_status, output, vm) =
             AptosSimulationVM::create_vm_and_simulate_signed_transaction(&txn, &state_view);
         let version = ledger_info.version();
 
         // Ensure that all known statuses return their values in the output (even if they aren't supposed to)
-        let exe_status = ExecutionStatus::convert_vm_status_for_simulation(vm_status.clone());
+        let exe_status = if vm.is_feature_enabled(FeatureFlag::REMOVE_DETAILED_ERROR_FROM_HASH)
```

Comment:
Let's move this into SimulationVM apis, this does not belong to simulation? All this code can be just added to the end of `create_vm_and_simulate_signed_transaction`